src/prometheus_async/aio/_decorators.py

```diff
@@ -16,11 +16,31 @@
 Decorators for asyncio.
 """
 
+from asyncio import Future
 from time import perf_counter
+from typing import Awaitable, Callable, Type, TypeVar, overload
 
+from typing_extensions import ParamSpec
 from wrapt import decorator
 
 
+P = ParamSpec("P")
+R = TypeVar("R", bound=Awaitable)
+T = TypeVar("T")
+
+
+@overload
+def time(
+    metric,
+) -> Callable[[Callable[P, R]], Callable[P, R]]:
+    ...
+
+
+@overload
+def time(metric, future: Future[T]) -> Awaitable[T]:
```

Comment:
I think this is what's breaking CI. Future isn't subscriptable…I think I've run into that problem myself once.